### PR TITLE
Set default object store directory via application.properties

### LINKED ISSUE #15255: Make quarkus.transaction-manager.enable-recovery configurable by Keycloak env-var
### Description

After adding the option to configure XA-transaction with default set to true, we experience the following warning since Keycloak 20.0.0.

```
WARN [io.quarkus.agroal.runtime.DataSources] (main) Datasource <default> enables XA but transaction recovery is not enabled. 
Please enable transaction recovery by setting quarkus.transaction-manager.enable-recovery=true, 
otherwise data may be lost if the application is terminated abruptly
```
This was introduced with #14375.

Can we please get the possibility to configure [quarkus.transaction-manager.enable-recovery](https://quarkus.io/guides/all-config#quarkus-narayana-jta_quarkus.transaction-manager.enable-recovery) as suggested in the warning so we don't need to create a `quarkus.properties` file?

Thank you for your incredible work!

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by sbrajchuk**
Set default directory name for the location of the transaction logs

Can be overriden by setting the [quarkus.transaction-manager.object-store-directory](https://quarkus.io/guides/all-config#quarkus-narayana-jta_quarkus.transaction-manager.object-store-directory) property in the quarkus.properties file or by setting the environment variable QUARKUS_TRANSACTION_MANAGER_OBJECT_STORE_DIRECTORY

Closes #15255

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]
@sbrajchuk @vmuzikar LGTM.

Perhaps the directory name is not so meaningful and `data/transaction/object-store` is better. But I'm not sure.

File: quarkus/runtime/src/main/resources/application.properties
Comment (vmuzikar) [reply quarkus/runtime/src/main/resources/application.properties:46]: @pedroigor Is that a correct format? ^^

Comment (sbrajchuk) [reply quarkus/runtime/src/main/resources/application.properties:46]: Tested on Windows and confirmed that the Linux separator functions properly. Your approach also works, but I believe your method appears to be more accurate.

**Review by vmuzikar — APPROVED** [review]
@sbrajchuk Thanks, LGTM.
Let's wait for an approval from @pedroigor before merging this.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@sbrajchuk Actually, one more thing. Could you please create and link a GH Issue to this, see our [contribution guide](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#contributing-to-keycloak).

**Review by pedroigor — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]

**Comment by pedroigor** [comment]
@sbrajchuk Thanks.

**Pull request merged**
